### ISSUE TITLE
Unable to install org-mode

### ISSUE THREAD

**Issue by ssbothwell** [raxod502/straight.el#854]
I'm on emacs 27.2 and  am getting this error when installing `org` via `straight-use-package`:
```
$ cd /home/solomon/.emacs.d/straight/repos/
$ git clone --origin origin --no-checkout https\://code.orgmode.org/bzg/org-mode.git /home/solomon/.emacs.d/straight/repos/org/ --no-single-branch

Cloning into '/home/solomon/.emacs.d/straight/repos/org'...
fatal: unable to update url base from redirection:
  asked for: https://code.orgmode.org/bzg/org-mode.git/info/refs?service=git-upload-pack
   redirect: https://git.savannah.gnu.org/cgit/emacs/org-mode.git/?service=git-upload-pack

[Return code: 128]
```

I tried replacing my entire `init.el` with:
```
(defvar bootstrap-version)
(let ((bootstrap-file
       (expand-file-name "straight/repos/straight.el/bootstrap.el" user-emacs-directory))
      (bootstrap-version 5))
  (unless (file-exists-p bootstrap-file)
    (with-current-buffer
        (url-retrieve-synchronously
         "https://raw.githubusercontent.com/raxod502/straight.el/develop/install.el"
         'silent 'inhibit-cookies)
      (goto-char (point-max))
      (eval-print-last-sexp)))
  (load bootstrap-file nil 'nomessage))

(setq package-enable-at-startup nil)
```
And got the same git error.

I also tried manually cloning the repo redirected URI and then straight gave me this error:
```
$ cd /home/solomon/.emacs.d/straight/repos/org/lisp/
$ git describe --match release\* --abbrev\=0 HEAD

[File error while Setting current directory]
```
And it staged changes erasing everything from my local repo.

**Comment by jeremyf** [comment]
As an interim, I went with:

```elisp
(use-package org :straight (org :type built-in))
```

This removed the latest work on org-mode, but at least shored up a similar problem.

**Comment by ashton314** [comment]
Can confirm. Is it related to the top posts here?

https://updates.orgmode.org/

**Comment by progfolio** [comment]
Are either of you on straight's develop branch?
see related: https://github.com/raxod502/straight.el/issues/847#issuecomment-927328515

**Comment by ssbothwell** [comment]
I am not on develop. Thank you for the temporary fix @jeremyf and i look forward to the new master release @progfolio

**Comment by galdor** [comment]
I have the same problem here, I will test the next master release as soon as it is available. Thank you so much for reacting that fast ! I was really impressed by the way Straight handled the situation, letting me update other packages without crashing or letting Emacs in a broken state.

**Comment by jeremyf** [comment]
I can confirm the following worked for me:

1. (Perhaps optional but was what I did) : removed ~/.emacs.d/straight
2. Restarted emacs (which straight then ran the bootstrap)
3. (Perhaps optional but was what I did) : In ~/.emacs.d/straight/repos/straight.el I checked out the develop version
4. Add `(setq straight-repository-branch "develop")` after my "bootstrap-version" declaration of straight.el
5. Restarted emacs

I suspect I would've only needed step 4 and 5, but this got things working for me.

**Comment by zamlz** [comment]
> I can confirm the following worked for me:
> 
>     1. In ~/.emacs.d/straight/repos/straight.el I checked out the develop version (instead of master).
> 
>     2. Add `(setq straight-repository-branch "develop")` after my "bootstrap-version" declaration of straight.el
> 
>     3. Restarted emacs
> 
> 
> [jeremyf/dotemacs@b8f72b3](https://github.com/jeremyf/dotemacs/commit/b8f72b308dbbeefb9cfcd53ffd69ccfd8e261f9b) contains the minor changes I made to get back to using straight to pull down org.

Thanks @jeremyf , can confirm this fix works as well!
but a quick question. Having done this, when should I switch back to the `master` branch?

**Comment by m4xxed** [comment]
Hi, this problem still persists for me. 

I have switched to the develop branch by doing this:

> I can confirm the following worked for me:
> 
>     1. In ~/.emacs.d/straight/repos/straight.el I checked out the develop version (instead of master).
> 
>     2. Add `(setq straight-repository-branch "develop")` after my "bootstrap-version" declaration of straight.el
> 
>     3. Restarted emacs

.. yet I still get the fatal warning that git is unable to update url base from redirection. 

I am on native-comp emacs-git from the AUR, if that is any help. I have also checked that straight.el repo is on the develop branch, just to be sure. 

I am honestly flummoxed by this problem and cannot solve it.

I am loading org using `(use-package org :config ...)`, so maybe therein lies the problem?

I also already deleted the straight and org repos and builds from my straight directory, and also tried deleting the whole straight directory, reinstalling completely. 

Any ideas would be appreciated!

**Comment by progfolio** [comment]
Please share the recipe you are using to install Org and the output of `straight-version`.

**Comment by m4xxed** [comment]
Output of `straight-version`:

```
prerelease (HEAD -> master, origin/master, origin/develop, origin/HEAD) af5437f 2021-09-27
```

Output of `straight-get-recipe` choosing `org`:
```
(org :type git :repo "https://git.savannah.gnu.org/git/emacs/org-mode.git" :local-repo "org" :depth full :pre-build (straight-recipes-org-elpa--build) :build (:not autoloads) :files (:defaults "lisp/*.el" ("etc/styles/" "etc/styles/*")))
```

I hope that is the information you meant.

**Comment by m4xxed** [comment]
I just found the error, after some tedious installation-deletion-reinstallation-testing:

The issue is that 

```
(use-package org-ref-cite-core
  :straight (org-ref-cite-core :host github :repo "jkitchin/org-ref-cite"))
```

tries to install `org-mode` from the outdated repository, I guess. 

I don't have enough experience to find a workaround for this immediately, but maybe I am not the only one facing this error. 
If anyone has a good suggestion as to how I can stop this package from installing `org-mode`, I'd be very grateful! For now, I have just disabled this part of my config...

Btw, I found the solution because you asked me to supply the `recipe` information! Turns out checking the docs for a while would have saved me hours of frustration and "debugging"...
